src/data-workspace/data-workspace.js

```diff
@@ -14,12 +14,14 @@ import { useDataValueSet } from '../shared/index.js'
 import styles from './data-workspace.module.css'
 import { EntryForm } from './entry-form.js'
 import { FinalFormWrapper } from './final-form-wrapper.js'
+import useCloseRightHandPanelOnSelectionChange from './use-close-right-hand-panel-on-selection-change.js'
 
 export const DataWorkspace = ({ selectionHasNoFormMessage }) => {
     const [{ dataSetId }] = useContextSelection()
     const { data } = useMetadata()
     const initialDataValuesFetch = useDataValueSet()
     const isValidSelection = useIsValidSelection()
+    useCloseRightHandPanelOnSelectionChange()
```

Comment:
this hook needed to be a level higher in the hierarchy otherwise it errors and doesn't close if the new selection is invalid

Reply:
Wondering if it'd make more sense to put this into the component in `right-hand-panel.js`. Just a thought, not a proposal

Reply:
> Wondering if it'd make more sense to put this into the component in `right-hand-panel.js`. Just a thought, not a proposal

yeah that makes sense .. moved it to `right-hand-panel.js`